tests/run-multitests: Make paths more deterministic.

andrewleech opened the pull request:
Make relative paths used internally to the mutlitest runner tool correctly relative to the script file itself, rather than current working directory. Allows running multitests from different directory as well as making some errors more consistent.

This path cleanup "improves" confusions like https://ptb.discord.com/channels/574275045187125269/1012732018183913513/1019879446066515968 by avoiding accidental imports of test scripts, so instead of `TypeError: 'module' object is not callable` when cpython imports micropython/consts.py test it now gives `ModuleNotFoundError: No module named 'micropython'` which makes more sense.

For reference I was using multitest with aioble tests and found it easier to be able to run from the target test folders. 
```
cd lib/micropython-lib/micropython/bluetooth/aioble/multitests

MICROPY_MICROPYTHON=~/micropython/ports/unix/build-ble/micropython ~/micropython/tests/run-multitests.py -t -i micropython,MICROPYBTUART=/dev/ttyACM0,MICROPYPATH=".frozen:$(pwd)/.." -i micropython,MICROPYBTUART=/dev/ttyACM1,MICROPYPATH=".frozen:$(pwd)/.." ble_write_order.py
```

I've also added some extra help messaging to the tool to make it easier to use / remember how to use.


---
This work was funded by Planet Innovation.

jimmo commented:
Thanks @andrewleech had been meaning to do this myself. LGTM!

andrewleech commented:
On a related note, I had a go at adding mpremote mount as a command line arg in multitest - thinking I might be able to mount and test aioble live on a device: https://github.com/micropython/micropython/commit/854095d984e6cb3db5eccb7720bd17b21e40c66a

Unfortunately it locks up in `self.pyb.exec_raw_no_follow(script)` as soon as anything's mounted. If I kill it and try to connect direct to repl, ctrl-c/d/a/z etc don't really work but eventually print out some eg. [00][02][ab] etc so I think it's stuck in file transfer, I'm assuming the `raw_no_follow` isn't allowing the correct intercept of file handling. I then switched to unix port and ble dongles - much faster!

dpgeorge commented on tests/run-multitests.py:
```diff
@@ -3,22 +3,36 @@
+    # accidentally importing tests like micropython/const.py
+    sys.path.pop(0)
+
+sys.path.insert(0, f"{test_dir}/../tools")
```
No other Python tool/test scripts in this repo use f-strings (yet, except mpremote), and since this is a very simple case please just do `test_dir + "/../tools"`.

Also 2 cases below.

andrewleech replied:
No worries, they've just become habit for me now :-)
That's fixed now thanks

pi-anl committed:
tests/run-multitests: Extend usage information.

pi-anl committed:
tests/run-multitests: Make paths more deterministic.

Allows running from different directory etc.

---
This work was funded by Planet Innovation.

dpgeorge commented:
Thanks.  Rebased and merged in 7589d86b6b6ea4232ff5614f3e73565529d4b3bb and 13c4470fd04981fce3fd29771e1d973febbe7d60
